homeassistant/components/refoss/device.py

```diff
@@ -0,0 +1,68 @@
+"""Entity for Refoss."""
+
+from __future__ import annotations
+
+from refoss_ha.controller.device import BaseDevice
+
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.dispatcher import async_dispatcher_connect
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, REFOSS_HA_SIGNAL_UPDATE_ENTITY
+
+
+class RefossEntity(Entity):
+    """Entity for refoss."""
+
+    _attr_has_entity_name = True
+    _attr_should_poll = False
+
+    def __init__(self, device: BaseDevice, channel: int) -> None:
+        """__init__."""
+        self._attr_unique_id = f"refoss_{device.uuid}_{channel}"
+        self._attr_name = (
+            device.dev_name if channel == 0 else f"{device.dev_name}-{channel}"
+        )
```

Comment:
```suggestion
        if channel == 0:
            self._attr_name = None
        else:
            self._attr_name = str(channel)
```
This will utilize has entity name = True better